ExporterDirectorPauseTest flaky

**Summary**

> Try to answer the following as best as possible

- How often does the test fail? saw it at least 2 times on PR jenkins builds and now on main
- Does it block your work? no
- Do we suspect that it is a real failure? no

**Failures**

> Outline known failure cases, e.g. a failed assertion and its stacktrace obtained from Jenkins

<details><summary>Example assertion failure</summary>
<pre>
java.lang.AssertionError: 
Expecting code not to raise a throwable but caught
  "java.util.concurrent.ExecutionException: Actor is closed
	at io.camunda.zeebe.scheduler.future.CompletableActorFuture.get(CompletableActorFuture.java:142)
	at io.camunda.zeebe.scheduler.future.CompletableActorFuture.get(CompletableActorFuture.java:109)
	at io.camunda.zeebe.scheduler.FutureUtil.join(FutureUtil.java:21)
	at io.camunda.zeebe.scheduler.future.CompletableActorFuture.join(CompletableActorFuture.java:198)
	at io.camunda.zeebe.broker.exporter.stream.ExporterDirectorPauseTest.lambda$canPauseAndResumeWithoutAnyExporter$2(ExporterDirectorPauseTest.java:101)
	at org.assertj.core.api.ThrowableAssert.catchThrowable(ThrowableAssert.java:63)
	at org.assertj.core.api.AssertionsForClassTypes.catchThrowable(AssertionsForClassTypes.java:892)
	at org.assertj.core.api.AssertionsForClassTypes.assertThatCode(AssertionsForClassTypes.java:863)
	at org.assertj.core.api.Assertions.assertThatCode(Assertions.java:1287)
	at io.camunda.zeebe.broker.exporter.stream.ExporterDirectorPauseTest.canPauseAndResumeWithoutAnyExporter(ExporterDirectorPauseTest.java:101)
	at jdk.internal.reflect.GeneratedMethodAccessor43.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:69)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater$1.execute(IdeaTestRunner.java:38)
	at com.intellij.rt.execution.junit.TestsRepeater.repeat(TestsRepeater.java:30)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:35)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:235)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:54)
Caused by: java.lang.RuntimeException: Actor is closed
	at io.camunda.zeebe.scheduler.ActorJob.failFuture(ActorJob.java:170)
	at io.camunda.zeebe.scheduler.ActorTask.failJob(ActorTask.java:261)
	at io.camunda.zeebe.scheduler.ActorTask.onClosed(ActorTask.java:255)
	at io.camunda.zeebe.scheduler.ActorTask.onAllJobsDone(ActorTask.java:194)
	at io.camunda.zeebe.scheduler.ActorTask.execute(ActorTask.java:159)
	at io.camunda.zeebe.scheduler.ActorThread.executeCurrentTask(ActorThread.java:106)
	at io.camunda.zeebe.scheduler.ActorThread.doWork(ActorThread.java:87)
	at io.camunda.zeebe.scheduler.ActorThread.run(ActorThread.java:198)
</pre>
</details>

**Hypotheses**

> List any hypotheses if you have one; can be ommitted

**Logs**

> If possible, provide more context here, e.g. standard output logs, link to build, etc.

https://ci.zeebe.camunda.cloud/job/camunda/job/zeebe/job/main/817/testReport/junit/io.camunda.zeebe.broker.exporter.stream/ExporterDirectorPauseTest(java-testrun)/Verify___Test___canPauseAndResumeWithoutAnyExporter__Flaky_Test_/

<details><summary>Logs from a local run</summary>
<pre>
08:48:42.788 [Broker-0-LogStream-1] [-zb-actors-4] DEBUG io.camunda.zeebe.logstreams - Configured log appender back pressure at partition 1 as AppenderVegasCfg{initialLimit=1024, maxConcurrency=32768, alphaLimit=0.7, betaLimit=0.95}. Window limiting is disabled
08:48:42.806 [Broker-0-LogStream-1] [-zb-actors-6] DEBUG io.camunda.zeebe.logstreams - Configured log appender back pressure at partition 1 as AppenderVegasCfg{initialLimit=1024, maxConcurrency=32768, alphaLimit=0.7, betaLimit=0.95}. Window limiting is disabled
08:48:43.038 [exporter] [-zb-fs-workers-1] DEBUG io.camunda.zeebe.broker.exporter - Recovering exporter from snapshot
08:48:43.039 [exporter] [-zb-fs-workers-1] DEBUG io.camunda.zeebe.broker.exporter - Recovered exporter 'exporter' from snapshot at lastExportedPosition -1
08:48:43.039 [exporter] [-zb-fs-workers-1] DEBUG io.camunda.zeebe.broker.exporter - Set event filter for exporters: ExporterEventFilter{acceptRecordTypes={COMMAND_REJECTION=false, EVENT=false, SBE_UNKNOWN=false, NULL_VAL=false, COMMAND=false}, acceptValueTypes={PROCESS_MESSAGE_SUBSCRIPTION=false, ERROR=false, PROCESS_INSTANCE_RESULT=false, MESSAGE_START_EVENT_SUBSCRIPTION=false, PROCESS_INSTANCE_MODIFICATION=false, SBE_UNKNOWN=false, TIMER=false, PROCESS=false, DEPLOYMENT_DISTRIBUTION=false, INCIDENT=false, CHECKPOINT=false, VARIABLE=false, DECISION_EVALUATION=false, DECISION_REQUIREMENTS=false, VARIABLE_DOCUMENT=false, MESSAGE_SUBSCRIPTION=false, DECISION=false, PROCESS_EVENT=false, MESSAGE=false, JOB_BATCH=false, PROCESS_INSTANCE=false, DEPLOYMENT=false, PROCESS_INSTANCE_CREATION=false, NULL_VAL=false, JOB=false}}
08:48:43.039 [exporter] [-zb-fs-workers-1] DEBUG io.camunda.zeebe.broker.exporter - Closed exporter director 'exporter'.
08:48:43.049 [Broker-0-LogStream-1] [-zb-actors-1] INFO  io.camunda.zeebe.logstreams - Close appender for log stream stream
08:48:43.050 [stream-write-buffer] [-zb-actors-1] DEBUG io.camunda.zeebe.dispatcher - Dispatcher closed
08:48:43.050 [Broker-0-LogStream-1] [-zb-actors-1] INFO  io.camunda.zeebe.logstreams - On closing logstream stream close 1 readers
08:48:43.050 [] [main] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-fs-workers'
08:48:43.050 [] [main] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-actors'
08:48:43.051 [] [-zb-fs-workers-1] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-fs-workers': closed successfully
08:48:43.051 [] [-zb-actors-4] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-actors': closed successfully
08:48:43.052 [Broker-0-LogStream-1] [-zb-actors-3] INFO  io.camunda.zeebe.logstreams - Close appender for log stream stream
08:48:43.053 [stream-write-buffer] [-zb-actors-3] DEBUG io.camunda.zeebe.dispatcher - Dispatcher closed
08:48:43.053 [Broker-0-LogStream-1] [-zb-actors-3] INFO  io.camunda.zeebe.logstreams - On closing logstream stream close 0 readers
08:48:43.053 [] [main] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-fs-workers'
08:48:43.053 [] [main] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-actors'
08:48:43.053 [] [-zb-fs-workers-0] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-fs-workers': closed successfully
08:48:43.054 [] [-zb-actors-4] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-actors': closed successfully

java.lang.AssertionError: 
Expecting code not to raise a throwable but caught
  "java.util.concurrent.ExecutionException: Actor is closed
	at io.camunda.zeebe.scheduler.future.CompletableActorFuture.get(CompletableActorFuture.java:142)
	at io.camunda.zeebe.scheduler.future.CompletableActorFuture.get(CompletableActorFuture.java:109)
	at io.camunda.zeebe.scheduler.FutureUtil.join(FutureUtil.java:21)
	at io.camunda.zeebe.scheduler.future.CompletableActorFuture.join(CompletableActorFuture.java:198)
	at io.camunda.zeebe.broker.exporter.stream.ExporterDirectorPauseTest.lambda$canPauseAndResumeWithoutAnyExporter$2(ExporterDirectorPauseTest.java:101)
	at org.assertj.core.api.ThrowableAssert.catchThrowable(ThrowableAssert.java:63)
	at org.assertj.core.api.AssertionsForClassTypes.catchThrowable(AssertionsForClassTypes.java:892)
	at org.assertj.core.api.AssertionsForClassTypes.assertThatCode(AssertionsForClassTypes.java:863)
	at org.assertj.core.api.Assertions.assertThatCode(Assertions.java:1287)
	at io.camunda.zeebe.broker.exporter.stream.ExporterDirectorPauseTest.canPauseAndResumeWithoutAnyExporter(ExporterDirectorPauseTest.java:101)
	at jdk.internal.reflect.GeneratedMethodAccessor43.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:69)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater$1.execute(IdeaTestRunner.java:38)
	at com.intellij.rt.execution.junit.TestsRepeater.repeat(TestsRepeater.java:30)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:35)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:235)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:54)
Caused by: java.lang.RuntimeException: Actor is closed
	at io.camunda.zeebe.scheduler.ActorJob.failFuture(ActorJob.java:170)
	at io.camunda.zeebe.scheduler.ActorTask.failJob(ActorTask.java:261)
	at io.camunda.zeebe.scheduler.ActorTask.onClosed(ActorTask.java:255)
	at io.camunda.zeebe.scheduler.ActorTask.onAllJobsDone(ActorTask.java:194)
	at io.camunda.zeebe.scheduler.ActorTask.execute(ActorTask.java:159)
	at io.camunda.zeebe.scheduler.ActorThread.executeCurrentTask(ActorThread.java:106)
	at io.camunda.zeebe.scheduler.ActorThread.doWork(ActorThread.java:87)
	at io.camunda.zeebe.scheduler.ActorThread.run(ActorThread.java:198)
"

	at io.camunda.zeebe.broker.exporter.stream.ExporterDirectorPauseTest.canPauseAndResumeWithoutAnyExporter(ExporterDirectorPauseTest.java:102)
	at jdk.internal.reflect.GeneratedMethodAccessor43.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:69)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater$1.execute(IdeaTestRunner.java:38)
	at com.intellij.rt.execution.junit.TestsRepeater.repeat(TestsRepeater.java:30)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:35)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:235)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:54)
</pre>
</details>

--- a/broker/src/test/java/io/camunda/zeebe/broker/exporter/stream/ExporterDirectorPauseTest.java
+++ b/broker/src/test/java/io/camunda/zeebe/broker/exporter/stream/ExporterDirectorPauseTest.java
@@ -22,6 +22,7 @@ import io.camunda.zeebe.protocol.impl.record.value.deployment.DeploymentRecord;
 import io.camunda.zeebe.protocol.record.intent.DeploymentIntent;
 import java.util.List;
 import java.util.Map;
+import org.awaitility.Awaitility;
 import org.junit.Before;
 import org.junit.Rule;
 import org.junit.Test;
@@ -94,6 +95,13 @@ public final class ExporterDirectorPauseTest {
   public void canPauseAndResumeWithoutAnyExporter() {
     // given
     activeExporter.startExporterDirector(List.of());
+
+    // when -- exporter is closed
+    // Wait for exporter to properly close before submitting pause/resume jobs.
+    // Needed to prevent flaky test https://github.com/camunda/zeebe/issues/10439
+    Awaitility.await()
+        .alias("Exporter is closed")
+        .until(() -> activeExporter.getDirector().getPhase().join() == ExporterPhase.CLOSED);
 
     // then
     assertThatCode(() -> activeExporter.getDirector().pauseExporting().join())

